src/core/Akka.Persistence/Persistence.cs

```diff
@@ -81,15 +79,13 @@ public string PersistenceId(IActorRef actor)
         public IActorRef SnapshotStoreFor(string snapshotPluginId)
         {
             var configPath = string.IsNullOrEmpty(snapshotPluginId) ? _defaultSnapshotPluginId.Value : snapshotPluginId;
-            Lazy<PluginHolder> pluginContainer;
-            var extensionIdMap = _snapshotPluginExtensionIds.Value;
-            if (!extensionIdMap.TryGet(configPath, out pluginContainer))
-            {
-                pluginContainer = new Lazy<PluginHolder>(() => CreatePlugin(configPath, _ => DefaultPluginDispatcherId), LazyThreadSafetyMode.ExecutionAndPublication);
-                _snapshotPluginExtensionIds.CompareAndSet(extensionIdMap, extensionIdMap.AddOrUpdate(configPath, pluginContainer));
-                return SnapshotStoreFor(snapshotPluginId);
-            }
```

Comment:
This part was the cause for race conditions in the old code.
there is a check if something does not exist, and if so, it adds an extension.. classic race condition semantics.

The new code fixes that.